### PR TITLE
A mule task for starting indexer nodes

### PR DESCRIPTION
**Pull request by bricerisingalgorand**
This is a mule task that allows us to start indexer instances

### CODE REVIEW

File: mule/util/algorand_util.py [hunk 1]
```diff
@@ -147,3 +147,52 @@ def goal(data_dir, kmd_dir, args, bin_dir=None):
...
+            '--pidfile', pid_file
+        ])
+
+    algorand_indexer(algorand_indexer_args, bin_dir, log_file_name)
+
+def build_indexer_postgress_connection_string(postgres):
+    postgress_connection_string = []
+    for field in postgres.items():
+        postgress_connection_string.append(f"{field[0]}={field[1]}")
+    return ' '.join(postgress_connection_string)
```
Comment (btoll) [inline mule/util/algorand_util.py:198]: These functions are so particular to a class that I think it would be better to have them in the class.

**Comment by btoll** [comment]
Overall I think it's fine, but if we're keeping the idea of making this a general purpose tool I'm not sure I'd approach it like this.  I think it would be better to pull out the algorand specific code into its own location and then simplly pass an object into `Start` that you need to start.

**Review by onetechnical — APPROVED** [review]
Overall the approach here looks fine to me, but I'm presuming you've tested it as well.

**Comment by bricerisingalgorand** [comment]
> Overall I think it's fine, but if we're keeping the idea of making this a general purpose tool I'm not sure I'd approach it like this. I think it would be better to pull out the algorand specific code into its own location and then simplly pass an object into `Start` that you need to start.

We've talked about breaking algorand code into a separate algorand-tasks repo, but since we're the only ones currently using mule I think it makes sense to keep them together for now. If we want we can have a story to break them apart. Mule supports outside task definitions pretty conveniently

**Comment by btoll** [comment]
> > Overall I think it's fine, but if we're keeping the idea of making this a general purpose tool I'm not sure I'd approach it like this. I think it would be better to pull out the algorand specific code into its own location and then simplly pass an object into `Start` that you need to start.
> 
> We've talked about breaking algorand code into a separate algorand-tasks repo, but since we're the only ones currently using mule I think it makes sense to keep them together for now. If we want we can have a story to break them apart. Mule supports outside task definitions pretty conveniently

Sounds good.

File: mule/util/algorand_util.py (hunk 1 above)
Comment (bricerisingalgorand) [reply mule/util/algorand_util.py:198]: I'm not totally sure what you mean. Are you saying these could go into the algorand.indexer.Start task? This function is going to be required later on by the indexer import features. I think it's ideal though to keep the responsibility of the task objects as slim as possible to keep the code more readable

**Review by egieseke — APPROVED** [review]
Looks good.

File: mule/util/algorand_util.py (hunk 1 above)
Comment (btoll) [reply mule/util/algorand_util.py:198]: No, they should be in the class itself.

**Pull request merged**